soc: nxp_imx: Add support for TRNG

aurel32 opened the pull request:
Add support for the TRNG device contained in the i.MX RT SoCs. It uses
the existing MCUX driver, and mostly consists in adding the Kconfig and
DTS entries.

Signed-off-by: Aurelien Jarno <aurelien@aurel32.net>

aurel32 committed:
soc: nxp_imx: Add support for TRNG

Add support for the TRNG device contained in the i.MX RT SoCs. It uses
the existing MCUX driver, and mostly consists in adding the Kconfig and
DTS entries.

Signed-off-by: Aurelien Jarno <aurelien@aurel32.net>

jhqian commented on drivers/entropy/entropy_mcux_trng.c:
```diff
@@ -11,6 +11,14 @@
+#ifndef TRNG0
+#define TRNG0 TRNG
+#endif
+
```
I don't think it's a good idea. After checking the code, I would suggest not to hard coding TRNG0 in this file, but to use something like  TRNG_BASE_PTRS[0] as a workaround, given the different macro of this base address.

aurel32 replied:
Thanks for your suggestion, I have just pushed a new version fixing that.

MaureenHelm replied:
Actually you should get the base pointer from device tree instead. Please make a separate commit to change the driver to use `DT_ENTROPY_MCUX_TRNG_BASE_ADDRESS` below, then a second commit to enable the driver on rt.

aurel32 replied:
I didn't realize that `TRNG` and `TRNG0` corresponded to the base address of the peripheral, thanks for the hint. I have just reworked the PR as you requested.

aurel32 committed:
drivers: entropy: mcux_trng: get the base address from the device tree

Instead of getting the base address from the MCUX headers, use the base
address from the device tree.

Signed-off-by: Aurelien Jarno <aurelien@aurel32.net>
